### DISCUSSION TITLE
Who's here for Day 6!?

### DISCUSSION THREAD

**Discussion by CarolineMorton** [CarolineMorton/AdventOfCode_WIR#8] [Ideas]
🎄🎄 Day 6 is here!  (and it's Friday!) 🎄🎄

Thank you to everyone who is taking part in this. I have seen some great solutions - and people rapidly improving their rust as they try new things out. Well done! 

Day 6 does not disappoint!. See you in the comments :-)

**Comment by AR2202** [comment]
[Day 6](https://github.com/AR2202/AdventOfCode_WIR/blob/main/day_06/src/main.rs)
Brute forcing Part 2 was a bit slow, but finished eventually.

**Reply by hollygrimm** [reply]
Great job, Annika!

**Reply by AR2202** [reply]
There is a clever trick that makes it much more efficient, I won't give it away here as a spoiler, but might refactor my solution to it at some point, so whoever wants to can check it out.

**Comment by hollygrimm** [comment]
Hi @CarolineMorton, I finished part one for day 6, but after struggling with part two, I've decided that it's no longer benefiting my Rust skills to complete these. I'm switching to https://www.rustfinity.com/advent-of-rust. It's much easier, but has more Rust-specific challenges.

**Reply by CarolineMorton** [reply]
Totally understand - please feel free to continue to share your solutions if helpful. And thanks for participating so far 🎄🎄

**Comment by kuhmil** [comment]
I am struggling with this one! Might have to see what you all come up with 🙂!

**Reply by AR2202** [reply]
If you have a specific question, you could ask here. Maybe someone can help.

**Reply by CarolineMorton** [reply]
Second this!

**Comment by em-baggie** [comment]
<a href="https://github.com/em-baggie/AdventOfCode_WIR/tree/main/day_06">My solution</a> to day 6 pt 1. Found this one a bit easier, used an enum to abstract away the logic relating to directions on the grid. Implemented some tests and custom errors.

**Reply by CarolineMorton** [reply]
great work Emma!

**Comment by DominikStefancik** [comment]
My [solution](https://github.com/DominikStefancik/AdventOfCode_WIR/tree/main/day_06) for the first part of day 6. Will work on the second part later.
